.vscode/settings.json

```diff
@@ -1,86 +1,165 @@
+  "editor.tabSize": 4,
+  "editor.wordWrap": "off",
+
   // ---------------------------------------------------------------------------
+
+  "[javascript][json][json5][jsonc][typescript][yaml]": {
+    "editor.defaultFormatter": "dbaeumer.vscode-eslint",
+    "editor.codeActionsOnSave": {
+      "source.fixAll.eslint": "explicit"
+    }
+  },
+
+  "[jsonc]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  },
+
+  "[liquid]": {
+    "editor.defaultFormatter": "Shopify.theme-check-vscode"
+  },
+
+  // "[toml]": {
+  //   "editor.codeActionsOnSave": {
+  //     "source.fixAll.dprint": "explicit"
+  //   },
+  //   "editor.defaultFormatter": "dprint.dprint"
+  // },
+
+  "[markdown]": {
+    "editor.codeActionsOnSave": {
+      "source.fixAll.markdownlint": "explicit"
+    },
+    "editor.formatOnSave": true,
+    // "editor.defaultFormatter": "esbenp.prettier-vscode"
+    "editor.defaultFormatter": "unifiedjs.vscode-remark"
+    // "editor.defaultFormatter": "DavidAnson.vscode-markdownlint"
+  },
+
+  // Render vertical rulers after a certain number of monospace characters.
+  "editor.rulers": [50, 72, 80, 100]   , // Uses multiple values for multiple rulers.
+
   // Terminal
   // The Linux profiles to present when creating a new terminal via the
   // terminal dropdown. When set, these will override the default detected
   // profiles. They are comprised of a `path` and optional `args`.
+  //////////////////////////////////////////////////////////////////////////////
+
+  // A set of terminal profile customizations for Linux that allows adding,
+  // removing, or changing how terminals are launched. Profiles are comprised of
+  // a mandatory `path` and optional `args`, and other presentation options.
   "terminal.integrated.profiles.linux": {
+    "bash": {
+      "path": "bash"
+    },
+    "zsh": {
+      "path": "zsh"
+    },
     "fish": {
-      "path": "/usr/bin/fish"
-    }
+      "path": "fish"
+    },
+    "tmux": {
+      "path": "tmux",
+      "icon": "terminal-tmux"
+    },
+    "pwsh": null
   },
-  "editor.formatOnSave": true,
-  // The default profile used on Linux. This setting will currently be ignored
-  // if either `terminal.integrated.shell.linux` or
-  // `terminal.integrated.shellArgs.linux` are set.
+
+  // The default terminal profile on Linux.
   "terminal.integrated.defaultProfile.linux": "fish",
-  // Auto-fix JS files with ESLint using our custom settings. Needs
-  // https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
-  "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
-  },
-  "[json5]": {
-    "editor.formatOnSave": true
+
+  // TODO(DerekNonGeneric):
+  // A set of terminal profile customizations for macOS which allows adding,
+  // removing, or changing how terminals are launched. Profiles are comprised of
+  // a mandatory `path` and optional `args`, and other presentation options.
+  "terminal.integrated.profiles.osx": {},
```

Comment:
```suggestion
  // A set of terminal profile customizations for macOS which allows adding,
  // removing, or changing how terminals are launched. Profiles are comprised of
  // a mandatory `path` and optional `args`, and other presentation options.
  "terminal.integrated.profiles.osx": {
  },
```

i have heard really bad things about Zsh (the default shell on macOS), but switching to bash would be weird, so we are going to make it fish shell if nobody else minds&hellip;

i am not planning to embark on an Oh My Zsh! side quest when we have the fish shell life well planned out and a beaten cow-path already

i am going to take the plunge myself now&hellip;

Reply:
```zsh
brew install fish
```

1. To change your login shell to fish:

2. Add the shell to `/etc/shells` with:

```zsh
echo /opt/homebrew/bin/fish | sudo tee -a /etc/shells
```

3. Change your default shell with:

```fish
chsh -s /opt/homebrew/bin/fish
```

Reply:
i used to use iTerm2, actually:

- https://iterm2.com/

Reply:
woah:
🔗 https://code.visualstudio.com/api/references/icons-in-labels#icon-contribution-point